### PR TITLE
Adding Big data Competency for all developers

### PR DESCRIPTION
**Pull request by yousefiparsa**
Added Big Data as a competency for:
- ML Dev Role
- Dev Role
- Senior Dev Role
- Dev Team Lead

### CODE REVIEW

File: competencies/big-data.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+- You have a good understanding of Statistics and Linear Algebra.
+
+- You have an acceptable knowledge of probabilities science as the quantitative measurement in big data.
+
+- You know what Cloud Computing is, what characteristics it has, and how it is related to big data.
+
+- You have an understanding of Hadoop ecosystem and its modules such as HDFS, Yarn, and MapReduce.
+
+- You have familiarity with Natural Language Processing, its basic concepts such as tokenization, lemmatization, bigram collocation detection, etc.
```
Comment (codeallthethingz) [inline competencies/big-data.md:25]: I don't think NLP goes in big data.  Connected, but not needed for this competency

File: competencies/big-data.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+- You know Big data's main characteristics; 6 Vs; what they are, what they mean, and how they are connected to each other.
+
+- You have a good understanding of big data related and main programming languages such as Python, R, and Java.
+
+- You are familiar with data warehousing technologies in big data, relational and non-relational database systems such as (but not limited to) MySQL, Oracle, DB2, and NoSQL, Hbase, HDFS, MongoDB, CouchDB, Cassandra, Teradata, etc.
+
+- You have an acceptable understanding and experience in working with computational frameworks in big data such as Apache Spark, Apache Storm, Apache Samza, Apache Flink and the classic MapReduce and Hadoop.
+
+- You have a good understanding of Statistics and Linear Algebra.
```
Comment (codeallthethingz) [inline competencies/big-data.md:17]: All of these "good understanding" are a bit too vague.  How will a manager systematically validate that you have a good understanding of linear algebra?

File: competencies/big-data.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Competency - Big Data
+
+Big data is a field that treats ways to analyze, systematically extract information from, or otherwise deal with data sets that are too large or complex to be dealt with by traditional data-processing application software. Data with many cases (rows) offer greater statistical power, while data with higher complexity (more attributes or columns) may lead to a higher false discovery rate. Big data challenges include capturing data, data storage, data analysis, search, sharing, transfer, visualization, querying, updating, information privacy and data source. Big data was originally associated with three key concepts: volume, variety, and velocity. When we handle big data, we may not sample but simply observe and track what happens. Therefore, big data often includes data with sizes that exceed the capacity of traditional software to process within an acceptable time and value.
+
+- Wikipedia: https://en.wikipedia.org/wiki/Big_data
+
+## How do you prove it?
```
Comment (codeallthethingz) [inline competencies/big-data.md:7]: I would add two sections of proof.
1. Built a big data processing system and show it to your manager.
2. Built something using the cloud vendor ETL products.

File: competencies/big-data.md (hunk 1 above)
Comment (yousefiparsa) [reply competencies/big-data.md]: Done!

File: competencies/big-data.md (hunk 2 above)
Comment (yousefiparsa) [reply competencies/big-data.md]: Changed!

**Pull request merged**